### PR TITLE
Markus/message history

### PR DESCRIPTION
**Pull request by kuosmark**
# Description

Chat page styling and components. Chats are fetched from backend. Chats can be moved from one category to another. A new message can be added to the frontend state.

Fixes [Chat page: Showing message history](https://trello.com/c/getsZ8fD/769-chat-page-showing-message-history) and [Chat page: Chat Lists (Active, Archived, Blocked)](https://trello.com/c/96xN8lMm/767-chat-page-chat-lists-active-archived-blocked)

## Why was the change made?

This is the basis for the main functionality of the app.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

It should be tested by e2e tests, but isn't yet.

- [ ] Unittest
- [ ] Cypress e2e -tests

# Caveats?

There are type errors in the temporary dummy function.

The ChatWIndow component needs to be further split into separate files.

Tests not yet added.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/features/Chat/ChatMenu.tsx [hunk 1]
```diff
@@ -1,26 +1,107 @@
...
 
 const ChatMenu = () => {
   const { t } = useTranslation('chat');
   const [showCategories, setShowCategories] = useState(false);
-  const [currentCategory, setCurrentCategory] =
-    useState<ChatCategory>('active');
-  const chats = [];
+
+  const activeCategory = useSelector(
+    (state: RootState) => state.chats.activeCategory,
```
[the remark this answers is not in the document]
Comment (kuosmark) [reply src/features/Chat/ChatMenu.tsx]: Done!

File: src/features/Chat/ChatMenu.tsx [hunk 2]
```diff
@@ -1,26 +1,107 @@
...
+  };
 
   return (
     <Container>
       <Row>
-        <Header variant="h1">{t(`menu.title.${currentCategory}`)}</Header>
+        <Header variant="h1">{t(`menu.title.${activeCategory}`)}</Header>
         <Buttons>
           {!!chats.length && (
             <IconButton
```
[the remark this answers is not in the document]
Comment (kuosmark) [reply src/features/Chat/ChatMenu.tsx]: I started the splitting work. It should still be improved though, especially with the ChatWindow file.

File: src/features/Chat/ChatWindow.tsx [hunk 3]
```diff
@@ -77,7 +136,22 @@ const ChatWindow = () => {
           </Buttons>
         )}
       </HeaderBar>
-      <ChatHistory></ChatHistory>
+      <ChatHistory ref={historyRef}>
+        {groupedMessages.map(group => (
+          <React.Fragment key={group.date}>
```
[the remark this answers is not in the document]
Comment (kuosmark) [reply src/features/Chat/ChatWindow.tsx]: Done!

File: src/features/Chat/ChatWindow.tsx [hunk 4]
```diff
@@ -7,26 +9,83 @@ import { Profile as ProfileIcon } from '@/components/Icons/Profile';
...
+        if (groupIndex === -1) {
+          acc.push({ date: messageDate, messages: [curr] });
+        } else {
+          acc[groupIndex].messages.push(curr);
+        }
+
+        return acc;
+      }, []);
+
+  return chat ? (
```
[the remark this answers is not in the document]
Comment (kuosmark) [reply src/features/Chat/ChatWindow.tsx:124]: I shortened it a bit, but there's still work.

**Comment by rottabonus** [comment]
Because this contains some type errors, closing this for now. New PR can be made using this branch, or then reopen this if there will be continued work

**Pull request closed without merging**